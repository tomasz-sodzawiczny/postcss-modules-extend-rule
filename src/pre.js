import chainPlugins from "postcss-plugin-chain";
import extendRule from "postcss-extend-rule";
import extendFrom from "./plugins/postcss-modules-extend-from";

const plugin = chainPlugins("postcss-modules-extend-rule/pre", [
  extendFrom,
  extendRule
]);

export default plugin;
