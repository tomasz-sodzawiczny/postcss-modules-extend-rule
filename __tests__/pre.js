import pre from "../src/pre";

const process = css =>
  pre.process(css, { from: undefined }).then(result => result.css);

describe("pre", () => {
  it("works with both extends and extends-from syntaxes", async () => {
    const css =
      ".blue {color: blue} .a { @extend .blue; } .b { @extend .red from 'red' }";

    expect(await process(css)).toMatchSnapshot();
  });
});
