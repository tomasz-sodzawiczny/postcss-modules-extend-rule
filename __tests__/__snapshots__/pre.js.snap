// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pre works with both extends and extends-from syntaxes 1`] = `
":import('red') {
  __extends_from_1__red: red
} .blue {color: blue} .a {
  color: blue
} .b { @postcss-modules-extend-class __extends_from_1__red}"
`;
